In [1]:
from textblob import TextBlob

In [39]:
import pandas as pd
import seaborn as sns
from langdetect import detect

ModuleNotFoundError: No module named 'langdetect'

### Reading in Data
Due to the size of the raw data from IMDb, several aspects of the data have been removed to further focus on specifics and so that Jupyter has less trouble handling the Data Frames

#### Actors
- Missing the birthYear value
- Birth years before 1900
- Primary Profession not listing "actor" or "actress"

#### Movies
- Removing all without a runtime
- Removing all that are not labeled as 'movie' (i.e. short films, television, etc)

In [6]:
actors_df = pd.read_csv('actors_data_1900.csv')
movies_df = pd.read_csv('movies_wruntime.csv')

In [14]:
movies_df.head()

tconst titleType                    primaryTitle  \
0  tt0000009     movie                      Miss Jerry   
1  tt0000147     movie   The Corbett-Fitzsimmons Fight   
3  tt0000574     movie     The Story of the Kelly Gang   
4  tt0000679     movie  The Fairylogue and Radio-Plays   
5  tt0001184     movie         Don Juan de Serrallonga   

                    originalTitle  isAdult startYear  runtimeMinutes  \
0                      Miss Jerry        0      1894              45   
1   The Corbett-Fitzsimmons Fight        0      1897              20   
3     The Story of the Kelly Gang        0      1906              70   
4  The Fairylogue and Radio-Plays        0      1908             120   
5         Don Juan de Serrallonga        0      1910              58   

                   genres  
0                 Romance  
1  Documentary,News,Sport  
3   Biography,Crime,Drama  
4       Adventure,Fantasy  
5         Adventure,Drama

### Data Cleaning
There are several entries that do not have a start year, so first we will clean it up by removing those. Then it is much easier to compare the years since they will now all be integer values.
Also since we are not working with any television shows, we can drop the endYear value and relabel the startYear column.

In [7]:
movies_df = movies_df[(movies_df.startYear !="\\N")].drop(columns = ['endYear'])

In an effort to remove any films listed that may not be relevant (very small productions, never released to a wide audience or during the turn of the century) generally have missing genre information so I will drop the entires without atleast one genre.

In [8]:
movies_df = movies_df[movies_df['genres'] !='\\N']

Now I am curious what the difference between the Original Title and the Primary Title, how often do they not match?

In [21]:
(movies_df['primaryTitle'] != movies_df['originalTitle']).value_counts()

False    116708
True      31315
dtype: int64

There are 31315 instances of the Primary Title and the Original Title not matching. After looking at the titles of the films, it seems that almost always the English title is considered the Primary Title and the Original Title is in the language of the origin country.

In [38]:
diffTitle_df = movies_df[movies_df['primaryTitle'] != movies_df['originalTitle']]
diffTitle_df

tconst titleType                                    primaryTitle  \
6        tt0001258     movie                           The White Slave Trade   
12       tt0001790     movie            Les Misérables, Part 1: Jean Valjean   
15       tt0001911     movie                                     Nell Gwynne   
16       tt0001964     movie                                   The Traitress   
17       tt0002026     movie                    Anny - Story of a Prostitute   
20       tt0002130     movie                                 Dante's Inferno   
21       tt0002153     movie                    The Great Circus Catastrophe   
22       tt0002186     movie                               The Flying Circus   
28       tt0002452     movie                     The Independence of Romania   
29       tt0002461     movie          The Life and Death of King Richard III   
30       tt0002514     movie                                Saved in Mid-Air   
40       tt0002844     movie       Fantômas: In the Shadow of the Guillotine   
42       tt0002898     movie                 Germinal; or, The Toll of Labor   
43       tt0002917     movie                         Brother Against Brother   
50       tt0003037     movie                      Fantomas: The Man in Black   
52       tt0003102     movie                                Love Everlasting   
55       tt0003131     movie                                     War Is Hell   
57       tt0003159     movie                 Les Misérables, Part 2: Fantine   
58       tt0003165     movie               Fantômas: The Dead Man Who Killed   
60       tt0003227     movie  The Girl from Abroad; or, The Great Underworld   
64       tt0003330     movie                      The Life of Richard Wagner   
67       tt0003419     movie                           The Student of Prague   
71       tt0003489     movie                        The Last Days of Pompeii   
73       tt0003565     movie                               Where Is Coletti?   
93       tt0003870     movie                                       Judge Not   
97       tt0003930     movie           Fantomas: The Mysterious Finger Print   
99       tt0003952     movie                  Fantômas: The False Magistrate   
107      tt0004022     movie                                   Julius Caesar   
108      tt0004026     movie                                       The Golem   
110      tt0004047     movie                                      Half Breed   
...            ...       ...                                             ...   
160826   tt1010052     movie                               Surveillance 24/7   
160827   tt1010055     movie                                       Yatterman   
160835  tt10101782     movie                                      Foodiverse   
160843  tt10102350     movie                                 2 Fingers Honey   
160847   tt1010248     movie                              Return of the Atom   
160851   tt1010265     movie                                   Fallen Angels   
160859  tt10102826     movie  Grutarally: The Return of 'Kid's Eating Trash'   
160864   tt1010333     movie                 Departing Osaka Station at 0:00   
160865   tt1010336     movie               13 Short Opuses for Silver Screen   
160866   tt1010341     movie                        Da-Yu: The Touch of Fate   
160874   tt1010398     movie                                The Suicide Song   
160875   tt1010399     movie                                 The Big Sellout   
160876   tt1010400     movie                                         The Sin   
160880   tt1010409     movie                          My DNA Says I Love You   
160882   tt1010414     movie         The Life and Work of Noriaki Tsuchimoto   
160883   tt1010415     movie                                   Lolita's Club   
160887   tt1010420     movie                                     Misbegotten   
160888   tt1010422     movie                                 Artificial Bump   
160891   tt1